api/src/main/java/ca/bc/gov/educ/api/grad/report/controller/v2/SchoolReportsController.java

```diff
@@ -0,0 +1,114 @@
+package ca.bc.gov.educ.api.grad.report.controller.v2;
+
+import ca.bc.gov.educ.api.grad.report.model.dto.SchoolReports;
+import ca.bc.gov.educ.api.grad.report.service.SchoolReportsService;
+import ca.bc.gov.educ.api.grad.report.util.*;
+import io.swagger.v3.oas.annotations.OpenAPIDefinition;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.info.Info;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import jakarta.validation.constraints.NotNull;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequestMapping(EducGradReportApiConstants.SCHOOL_REPORTS_ROOT_MAPPING)
+@OpenAPIDefinition(info = @Info(title = "API for School Reports endpoints.", description = "This API is for reading and updating endpoints.", version = "2"), security = {@SecurityRequirement(name = "OAUTH2", scopes = {"READ_GRAD_STUDENT_REPORT_DATA","UPDATE_GRAD_STUDENT_REPORT_DATA"})})
+public class SchoolReportsController {
+  private static final Logger logger = LoggerFactory.getLogger(SchoolReportsController.class);
+
+  SchoolReportsService service;
+  GradValidation validation;
+  ResponseHelper response;
+
+  @Autowired
+  public SchoolReportsController(SchoolReportsService service, GradValidation validation, ResponseHelper response) {
+      this.service = service;
+      this.validation = validation;
+      this.response = response;
+  }
+
+  @GetMapping(EducGradReportApiConstants.SEARCH_MAPPING)
+  @PreAuthorize(PermissionsConstants.READ_GRADUATION_STUDENT_REPORTS)
+  @Operation(summary = "Search School Reports", description = "Search for school reports with optional filters", tags = {"Reports"})
+  @ApiResponses(value = {
+          @ApiResponse(responseCode = "200", description = "OK"),
+          @ApiResponse(responseCode = "400", description = "Bad request")})
```

Comment:
I don't see any validation mechanism that would result in a 400. But in this case, as it is a search and not an object payload, we can do away with the 400.

Reply:
400 is for the uuid